### PR TITLE
Postgres native types validations

### PR DESCRIPTION
**Pull request by carmenberndt**
- validates Serial data types using a number as a default value

related: https://www.notion.so/prismaio/Compatibilities-with-Prisma-attributes-6e0902265bc94b8f88ccb47d780fcccb

### CODE REVIEW

File: libs/datamodel/connectors/sql-datamodel-connector/src/postgres_datamodel_connector.rs [hunk 1]
```diff
@@ -140,6 +141,24 @@ impl Connector for PostgresDatamodelConnector {
                     _ => {}
                 }
             }
+            if native_type_name == SMALL_SERIAL_TYPE_NAME
+                || native_type_name == SERIAL_TYPE_NAME
+                || native_type_name == BIG_SERIAL_TYPE_NAME
+            {
+                if let Some(default_value) = field.default_value() {
+                    match default_value {
+                        DefaultValue::Single(_) => {
```
Comment (tomhoule) [inline libs/datamodel/connectors/sql-datamodel-connector/src/postgres_datamodel_connector.rs:150]: nitpick: you can collapse the match to reduce the nesting here, something like:

`if let Some(DefaultValue::Single(_)) = field.default_value() { return Err(...) }`

File: libs/datamodel/connectors/sql-datamodel-connector/src/postgres_datamodel_connector.rs [hunk 2]
```diff
@@ -140,6 +141,24 @@ impl Connector for PostgresDatamodelConnector {
                     _ => {}
                 }
             }
+            if native_type_name == SMALL_SERIAL_TYPE_NAME
```
Comment (dpetrick) [inline libs/datamodel/connectors/sql-datamodel-connector/src/postgres_datamodel_connector.rs:144]: I've seen some larger ifs in your PRs, if you like, you can do `matches!` as well to not have to do `native_type_name ==` all the time:
```rust
if matches!(native_type_name, SMALL_SERIAL_TYPE_NAME | SERIAL_TYPE_NAME | BIG_SERIAL_TYPE_NAME) {
 // ...
}
```
Not really a requirement, more a flavor question.

File: libs/datamodel/core/tests/types/postgres_native_types.rs [hunk 3]
```diff
@@ -1,6 +1,21 @@
...
+#[test]
+fn should_fail_on_serial_data_types_with_number_default() {
+    fn error_msg(type_name: &str) -> String {
+        format!(
+            "Sequential native type {} of Postgres must not have a static default value.",
+            type_name
+        )
+    }
+
+    for tpe in vec!["SmallSerial", "Serial", "BigSerial"] {
```
Comment (tomhoule) [inline libs/datamodel/core/tests/types/postgres_native_types.rs:14]: nitpick: `&["SmallSerial", "Serial", "BigSerial"]` would also work

**Review by tomhoule — APPROVED** [review]
looks good! Feel free to ignore the suggestions, it's very small things.

**Review by dpetrick — APPROVED** [review]

File: libs/datamodel/core/tests/types/postgres_native_types.rs (hunk 3 above)
Comment (dpetrick) [reply libs/datamodel/core/tests/types/postgres_native_types.rs:14]: Beat me to it :) Tom's suggestion doesn't allocate memory at runtime, which is nice.

File: libs/datamodel/connectors/sql-datamodel-connector/src/postgres_datamodel_connector.rs (hunk 2 above)
Comment (carmenberndt) [reply libs/datamodel/connectors/sql-datamodel-connector/src/postgres_datamodel_connector.rs:144]: Oh that's neat! Didn't know about that.

**Pull request merged**